modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml

```diff
@@ -17,6 +17,23 @@
       description: |
         Check what's going on: `kubectl -n {{ $labels.namespace }} logs {{ $labels.pod }}`
 
+  - alert: CiliumAgentNotFountHealthEndpoints
+    expr: (count by (namespace,pod) (cilium_unreachable_health_endpoints) OR count by (namespace,pod) (cilium_endpoint_state)) != 1
+    for: 5m
+    labels:
+      d8_module: cni-cilium
+      d8_component: agent
+      severity_level: "4"
+      tier: cluster
+    annotations:
+      plk_protocol_version: "1"
+      plk_markup_format: "markdown"
+      plk_grouped_by__main: CiliumAgentMalfunctioning,prometheus=deckhouse,tier=~tier,kubernetes=~kubernetes
+      plk_create_group_if_not_exists__main: CiliumAgentMalfunctioning,prometheus=deckhouse,tier=~tier,kubernetes=~kubernetes
+      summary: Some node's health endpoints are not reachable by agent {{ $labels.namespace }}/{{ $labels.pod }}.
```

Comment:
Can we extract which endpoints are unreachable by agent?

Reply:
Only by making a rule for each metric.